File: dbpro/src/main/java/feat/payment/MoviePaymentDAO.java
package feat.payment;

import oracle.connect.JDBCManager;
import org.apache.commons.lang3.StringUtils;

import java.util.List;
import java.util.Map;

public class MoviePaymentDAO {

	private static MoviePaymentDAO instance;

	private MoviePaymentDAO() {
	}

	public static MoviePaymentDAO getInstance() {
		if (instance == null) {
			instance = new MoviePaymentDAO();
		}
		return instance;
	}

	public List<MoviePaymentDTO> selectMoviePayments() {
		final String query = "SELECT " +
			"    MOVIE_PAYMENT_CODE, " +
			"    MYUSER_ID, " +
			"    MOVIE_CODE, " +
			"    PAYMENT_CODE, " +
			"    PAYMENT_DATE " +
			"FROM " +
			"    MOVIE_PAYMENT ";

		return JDBCManager
			.getInstance()
			.queryForList(query, MoviePaymentDTO.class);
	}

	public List<Map<String, Object>> selectMyMoviePayments(String id) {

		final String query = "SELECT " +
			"    mp.movie_payment_code, " +
			"    mp.myuser_id, " +
			"    m.movie_title, " +
			"    m.movie_age, " +
			"    p.payment_way, " +
			"    mp.payment_date " +
			"FROM " +
			"    movie_payment mp " +
			"JOIN " +
			"    movie m ON mp.movie_code = m.movie_code " +
			"JOIN " +
			"    payment p ON mp.payment_code = p.payment_code " +
			"WHERE " +
			"    mp.myuser_id = " + StringUtils.wrap(id, "'");

		return JDBCManager
			.getInstance()
			.queryForMaps(query, new String[] {"mp.movie_payment_code",
				"mp.myuser_id",
				"m.movie_title",
				"m.movie_age",
				"p.payment_way",
				"mp.payment_date"
			});
	}
}

File: dbpro/src/main/java/kr/ac/sejong/feat/payment/MoviePaymentDAO.kt
package kr.ac.sejong.feat.payment

import kr.ac.sejong.config.JDBCManager
import org.apache.commons.lang3.StringUtils

class MoviePaymentDAO private constructor() {
    companion object {
        @JvmStatic
        var instance: MoviePaymentDAO? = null
            get() {
                if (field == null) {
                    field = MoviePaymentDAO()
                }
                return field
            }
            private set
    }

    fun selectMoviePayments(): List<MoviePaymentDTO> {
        val query = """ 
            SELECT
                MOVIE_PAYMENT_CODE,
                MYUSER_ID,
                MOVIE_CODE,
                PAYMENT_CODE,
                PAYMENT_DATE
            FROM
                MOVIE_PAYMENT
        """

        return JDBCManager
            .getInstance()
            .queryForList(query, MoviePaymentDTO::class.java)
    }

    fun selectMyMoviePayments(id: String?): List<Map<String, Any>> {
        val query = """
            SELECT
                mp.movie_payment_code,
                mp.myuser_id,
                m.movie_title, 
                m.movie_age,
                p.payment_way,
                mp.payment_date
            FROM
                movie_payment mp
            JOIN
                movie m ON mp.movie_code = m.movie_code
            JOIN
                payment p ON mp.payment_code = p.payment_code
            WHERE
                mp.myuser_id = $id
        """

        return JDBCManager
            .getInstance()
            .queryForMaps(query, arrayOf("mp.movie_payment_code",
                "mp.myuser_id",
                "m.movie_title",
                "m.movie_age",
                "p.payment_way",
                "mp.payment_date"
            ))
    }
}